ui/pages/snap-account-redirect/components/url-display-box.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { SnapAccountRedirectProps } from '../snap-account-redirect';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderRadius,
+  BorderColor,
+  Display,
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { Box, Text } from '../../../components/component-library';
+import RedirectUrlIcon from './redirect-url-icon';
+
+const UrlDisplayBox = ({ url }: Pick<SnapAccountRedirectProps, 'url'>) => {
+  return (
+    <Box
+      display={Display.InlineFlex}
+      backgroundColor={BackgroundColor.backgroundDefault}
+      alignItems={AlignItems.center}
+      borderWidth={1}
+      borderRadius={BorderRadius.SM}
+      borderColor={BorderColor.borderDefault}
+      paddingRight={4}
+    >
+      <Text
+        data-testid="snap-account-redirect-url-display-box"
+        padding={2}
+        variant={TextVariant.bodyMd}
+        color={TextColor.primaryDefault}
+      >
+        {url}
+      </Text>
+      <RedirectUrlIcon url={url} />
+    </Box>
+  );
+};
+
+export default React.memo(UrlDisplayBox);
```

Comment:
Is this component meant to look like a TextField? I'm trying to understand the intention behind displaying a link in a TextField?

<img width="651" alt="Screenshot 2023-10-12 at 1 58 38 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/6aba6414-789d-4110-a5ea-f1fcd6b7d1b7">

In the design it uses a TextField so the intention would be for the value to be editable or readonly which should allow the user to copy the value easily, I haven't seen a pattern of a TextField being used for a link?

<img width="1022" alt="Screenshot 2023-10-12 at 1 59 07 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/f7d73ecf-1e4b-441b-a01b-a2490a170305">

Reply:
Agreed, I think we should have another design pattern for (external) links. Or at the very least make it a `disabled` text field so it doesn't look editable

Reply:
The url component right now is just a `Text` component and not a `TextField`. I agree, we should have another pattern for external links.

Reply:
I migrated from a TextField to a normal text component but I agree we should find a better more standard pattern. This is something that should happen at a later date IMO.